src/microbe_stage/editor/MicrobeEditor.cs

```diff
@@ -1233,6 +1266,39 @@ private void UpdateEditor(float delta)
         {
             CurrentOrganelleCost = 0;
         }
+
+        var vector = Vector3.Zero;
+        if (cameraMovementLeft)
+            vector += Vector3.Left;
+        if (cameraMovementRight)
+            vector += Vector3.Right;
+        if (cameraMovementUp)
+            vector += Vector3.Forward;
+        if (cameraMovementDown)
+            vector += Vector3.Back;
+
+        // Apply camera movement
+        camera.ObjectToFollow.Translation += vector.Normalized() * camera.CameraHeight * delta;
+    }
+
+    private void HandleMovement()
+    {
+        if (mousePanningStart == null)
+        {
+            cameraMovementLeft = Input.IsActionPressed("e_pan_left");
+
+            cameraMovementRight = Input.IsActionPressed("e_pan_right");
+
+            cameraMovementDown = Input.IsActionPressed("e_pan_down");
+
+            cameraMovementUp = Input.IsActionPressed("e_pan_up");
+        }
+        else
+        {
+            // ReSharper disable once PossibleInvalidOperationException
+            var direction = mousePanningStart.Value - camera.CursorWorldPos;
+            camera.ObjectToFollow.Translation += direction;
```

Comment:
Why is this in the input processing?
You aren't checking if the mouse was moved or not. I think you should put this code in _Process.